network.go

```diff
@@ -903,6 +903,11 @@ func (n *network) updateSvcRecord(ep *endpoint, localEps []*endpoint, isAdd bool
 			ipv6 = iface.AddressIPv6().IP
 		}
 
+		sb := ep.Sandbox()
```

Comment:
Being an internal alias, I was actually trying to avoid making the short-id to show up as an alias at docker level.
But, you are correct. This will not work in the remote hosts (though it will work in local hosts) n an overlay network.

I guess the best fix now is to make it an explicit `alias` from docker/docker. so that it will work seamlessly for all the network drivers.